File: app/src/main/java/com/example/androidpracticeview/adapter/MainAdapter.java
package com.example.androidpracticeview.adapter;

import androidx.annotation.Nullable;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;
import com.example.androidpracticeview.R;
import com.example.androidpracticeview.bean.TypeBean;

import java.util.List;

/**
 *
 */

public class MainAdapter extends BaseQuickAdapter<TypeBean, BaseViewHolder> {
    public MainAdapter( @Nullable List<TypeBean> data) {
        super(R.layout.adapter_item_main, data);
    }
    
    @Override
    protected void convert(BaseViewHolder baseViewHolder, TypeBean typeBean) {
        baseViewHolder.setText(R.id.title_tv, typeBean.getTitle());
    }
}

File: app/src/main/java/com/example/androidpracticeview/adapter/MainAdapter.kt
package com.example.androidpracticeview.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.androidpracticeview.R
import com.example.androidpracticeview.bean.TypeBean

/**
 * <pre>
 *      @author : Joseph
 *      e-mail  : 913870737@qq.com
 *      date    : 2022年1月21日
 *      desc    :
 *      version : 1.0
 * </pre>
 */
class MainAdapter(
    val dataList: List<TypeBean>,
    private val context: Context
    ) : RecyclerView.Adapter<MainAdapter.MainAdapterViewHolder>() {
    class MainAdapterViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var textView: TextView = view.findViewById(R.id.title_tv)
    }

    override fun onCreateViewHolder(parent: ViewGroup, p1: Int): MainAdapterViewHolder{
        return MainAdapterViewHolder(LayoutInflater.from(context).inflate(R.layout.adapter_item_main, parent, false))
    }

    override fun onBindViewHolder(viewHolder: MainAdapterViewHolder, position: Int) {
        viewHolder.textView.text = dataList[position].toString()
        viewHolder.itemView.setOnClickListener {
            setOnItemClickListener?.invoke(position)
        }
    }

    override fun getItemCount() = dataList.size

    var setOnItemClickListener:((position:Int)->Unit)? = null
}